### DISCUSSION TITLE
Get Input value from User and save it to a variable for later use

### DISCUSSION THREAD

**Discussion by Ms-Ajith** [chriskiehl/Gooey#839] [General]
Hello,
Iam creating a gui using Gooey and would like to save the input from user( a directory path) to use the file for further processing. How can I achieve this?

Thanks and Regards

**Comment by chriskiehl** [comment]
Tough to say without more information about what you're trying to accomplish or a code sample, but everything entered by the user is returned from the `parse_args()` call and available for use throughout your program 

```
parser = GooeyParser()
parser.add_argument('foo', widget='DirChooser')
args = parse_args()

args.foo  #<-- whatever directory the user picked 

arg.s
```

**Comment by Ms-Ajith** [comment]
Thanks alot for the help. This is working for me now. I also want to ask one more thing, Is it possible to make interactive gui with Gooey?
    curl_parser.add_argument('add',metavar='CHOOSE A Thing ',widget ='CheckBox')
If the user clicks the Checkbox, I need a filebrowser widget to add to the existing gui. Is it possible to make it work?

Also I would like to know whether if we use optional arguments in an argument, cant we add a variable to call back?
    curl_parser.add_argument("-o",'add',metavar='CHOOSE A Thing ',widget ='CheckBox')
This command is giving me error.
ValueError: invalid option string 'add': must start with a character '-'

Thanks

**Reply by chriskiehl** [reply]
Not currently. The alpha has some functionality that's moving in that direction, but it's still under baked at the moment. 

That said, you could probably get pretty close to what you're after by using a MutuallyExclusiveGroup. That would give you the behavior you're after where when a user clicks a button (in this case, a radio toggle), they can interact with a new field which was previously disabled.

**Reply by Ms-Ajith** [reply]
Thanks alot for the reply. I will look in to the MutuallyExclusiveGroup :)
